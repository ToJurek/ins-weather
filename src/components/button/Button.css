.refresh-button {
    background-color: hsla(0, 0%, 5%, 0.71);
    border-radius: 14px;
    margin-top: 5%;
    width: 90%;
    border: 1px solid #3c3c3c;
    display: inline-block;
    cursor: pointer;
    color: #f0f5f5;
    font-size: 17px;
    padding: 16px 31px;
    text-decoration: none;
    text-shadow: 0px 1px 0px #f0f5f5;
    outline: none;
    font-weight: bold;
}

.refresh-button:hover {
    background-color: #f0f5f5;
    color: black;
}

.refresh-button:active {
    position: relative;
    top: 1px;
}
