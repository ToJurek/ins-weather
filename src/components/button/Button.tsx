import React from 'react';
import './Button.css';

interface Props {
    name: string
    handleClick: any
}

function Button(props: Props) {
    return (<button className={'refresh-button'} onClick={() => props.handleClick()}>{props.name}</button>)
}

export default Button
