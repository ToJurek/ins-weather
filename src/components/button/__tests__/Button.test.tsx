import React from "react";
import Enzyme, {mount, shallow} from 'enzyme';
import Button from "../Button";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

Enzyme.configure({adapter: new Adapter()});

describe('button', function () {
    let wrapper: any
    let mockFunction: any
    beforeEach(() => {
        mockFunction = jest.fn()
        wrapper = mount(<Button name={'Button-name'} handleClick={mockFunction}/>)
    })
    test('should match to snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    test('should mock function be called', () => {
        wrapper.simulate('click');
        expect(mockFunction.mock.calls.length).toEqual(1)
    })

    test('should render button name correctly', () => {
        wrapper.update()
        expect(wrapper.find('button').text()).toBe('Button-name')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});
