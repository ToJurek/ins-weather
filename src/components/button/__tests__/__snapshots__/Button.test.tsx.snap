// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should match to snapshot 1`] = `
<Button
  handleClick={[MockFunction]}
  name="Button-name"
>
  <button
    className="refresh-button"
    onClick={[Function]}
  >
    Button-name
  </button>
</Button>
`;

exports[`button should render button name correctly 1`] = `
<Button
  handleClick={[MockFunction]}
  name="Button-name"
>
  <button
    className="refresh-button"
    onClick={[Function]}
  >
    Button-name
  </button>
</Button>
`;
